sensor/kubernetes/listener/resources/hash_reconciliation.go

```diff
@@ -0,0 +1,75 @@
+package resources
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/stringutils"
+)
+
+// ResourceStoreReconciler handles sensor-side reconciliation using in-memory store
+type ResourceStoreReconciler struct {
+	storeProvider *InMemoryStoreProvider
+}
+
+// NewResourceStoreReconciler builds ResourceStoreReconciler for sensor-side reconciliation
+func NewResourceStoreReconciler(storeProvider *InMemoryStoreProvider) *ResourceStoreReconciler {
+	return &ResourceStoreReconciler{storeProvider: storeProvider}
+}
+
+// ProcessHashes orchestrates the sensor-side reconciliation after a reconnect. It returns a slice of resource IDs that
+// should be deleted in Central to keep the state of Sensor and Central in sync.
+func (hr *ResourceStoreReconciler) ProcessHashes(h map[string]uint64) []central.MsgFromSensor {
```

Comment:
Let's update the signature and the interface already with a new key type `Key` as per the conversation with Luis. 

You can assume a key interface like:
```go
type Key interface {
        GetID() string
        GetType() reflect.Type
}
```